tests/Browser/FilterPostsTest.php

```diff
@@ -0,0 +1,165 @@
+                    ->assertSelected('select', 'accepted')
+                    ->press('Apply Filters')
+                    ->pause(3000)
+                    ->assertDontSee('@activeRejectButton', 'Reject')
+                    ->assertSeeIn('@activeAcceptButton', 'Accept');
+
+        });
+    }
+
+    /**
+     * Test filtering by tag.
+     */
+    public function testFilterByTag()
+    {
+        // Create a signup and an associated post with a 'accepted' status
+        // so the filter will show results.
+        $signup = factory(Signup::class)->create();
+        $post = $this->createAssociatedPostWithStatus($signup, 'accepted');
+        $post->tag('Good Photo');
+
+        $this->browse(function (Browser $browser) use ($signup) {
+            $browser->visit(new HomePage)
+                    // We're already logged in from the first test.
+                    ->assertPathIs('/campaigns')
+                    ->visit('/campaigns/' . $signup->campaign_id)
+                    ->on(new CampaignSinglePage($signup->campaign_id))
+                    ->assertSee('Post Filters')
+                    ->select('select', 'accepted')
+                    ->assertSelected('select', 'accepted')
+                    ->press('Good Photo')
+                    ->press('Apply Filters')
+                    ->pause(3000)
+                    ->assertDontSee('@activeTagButton', 'Good Quote')
+                    ->assertSeeIn('@activeTagButton', 'Good Photo');
+        });
+    }
+
+    /**
+     * Test no results page.
+     */
+    public function testNoResultsPage()
+    {
+        // Create a signup and an associated post with a 'accepted' status
+        // so the filter will show results.
+        $signup = factory(Signup::class)->create();
+        $post = $this->createAssociatedPostWithStatus($signup, 'accepted');
+
+        $this->browse(function (Browser $browser) use ($signup) {
+            $browser->visit(new HomePage)
+                    // We're already logged in from the first test.
+                    ->assertPathIs('/campaigns')
+                    ->visit('/campaigns/' . $signup->campaign_id)
+                    ->on(new CampaignSinglePage($signup->campaign_id))
+                    ->assertSee('Post Filters')
+                    ->select('select', 'accepted')
+                    ->assertSelected('select', 'accepted')
+                    ->press('Good For Storytelling')
+                    ->press('Apply Filters')
+                    ->pause(3000)
+                    ->assertSee('There are no results!');
+        });
+    }
+
+    /**
+     * Helper function to create a post with a specific status and associate it with a signup.
+     */
+    private function createAssociatedPostWithStatus($signup, $status)
+    {
+        $post = $signup->posts()->save(factory(Post::class)->make(['status' => $status]));
+        $post->campaign_id = $signup->campaign_id;
+        $post->save();
+
```

Comment:
this was a little strange but for some reason, the post wasn't getting the `campaign_id` when it was created. I had to fill this in and save it manually here.

Reply:
I tried to update the model factory by doing:
```
$factory->define(Post::class, function (Generator $faker) {
    $uploadPath = $faker->file(storage_path('fixtures'));
    $upload = new UploadedFile($uploadPath, basename($uploadPath), 'image/jpeg');
    $url = app(AWS::class)->storeImage($upload, $faker->unique()->randomNumber(5));
    $signup = factory(Signup::class)->create();
    return [
        'signup_id' => $signup->campaign_id,
        'url' => $url,
        'caption' => $faker->sentence(),
        'source' => $faker->randomElement(['phoenix-oauth', 'phoenix-next']),
        'remote_addr' => $faker->ipv4,
        'status' => 'pending',
        'campaign_id' => $signup->campaign_id,
    ];
});
```

but for some reason in the test, when I `dd` the `$signup->campaign_id` and the `$post->campaign_id` they were different. So reverting back tot his helper method.